### PR TITLE
ApiWrapper methods PUT by id: qualifying & matching questions

### PR DESCRIPTION
**Pull request by arshiamasih**
Addresses issue #234 

**Splits the generic putHostResponse method into two methods:**
- putMatchingResponse
- putShowstopperResponse

**Updates in files:**
- ApiWrapper
- host-context
- MatchingQuestion
- ShowstopperQuestion

Note: The proposed model is commented out for now as the current `HostResponse` model is in use in the data-context. A separate model could be created.
```
{ 
  email: string
  responseValues: string | Array<number> | number
}
```
Disregard any commented out code unrelated to issue #234

### CODE REVIEW

File: app/src/data/ApiWrapper.ts [hunk 1]
```diff
@@ -123,6 +125,12 @@ export class ApiWrapper {
         this.hostQuestionsResponse = new Fetcher<string>(
             `/api/v1/hostRegisterQuestions/${id}`
         )
+        this.hostShowstopperResponse = new Fetcher<string>(
```
Comment (tylerthome) [inline app/src/data/ApiWrapper.ts:128]: Please remove the "/api" prefix from these URLs, as that value is prepended by `Fetcher`'s constructor to potentially support other environments and API versions.

**Review by tylerthome — CHANGES_REQUESTED** [review]
Looks good so far! Aside form the requested change, this is ready to merge.  The use of `Promise.all` will yield better concurrency, and these resource types are similar enough that destructuring the 2D array in the reducer seems reasonable, but may merit some explanatory comments near that logic..

**Comment by arshiamasih** [comment]
@tyler any suggestions on a better approach to this? I used the 'any' type on the payload to follow the approach I used, and I don't know how to resolve or improve that either.
_and these resource types are similar enough that destructuring the 2D array in the reducer seems reasonable, but may merit some explanatory comments near that logic.._

**Comment by arshiamasih** [comment]
the `HostResponse ` interface is being used in data-context in what I believe is hard-coded matching results. 

Should I for now create a separate model for the qualifying and matching host response so as not to potentially break anything?

Also where will be the user email be accessible from?

**Comment by tylerthome** [comment]
> @tylerthome any suggestions on a better approach to this? I used the 'any' type on the payload to follow the approach I used, and I don't know how to resolve or improve that either.

The simplest way to keep the current pattern while avoiding the use of offset indexing for the destructure would be to assign directly into named collections, i.e. `const [myShowstopperQuestions, myMatchingQuestions] = Promise.all([...])` and instead of modeling the `action.payload` as an array, use an interface with these collections as named properties. Then, the reducer can query each by name without typecasting from `any` or using integer indices.

> the HostResponse interface is being used in data-context in what I believe is hard-coded matching results.
> 
> Should I for now create a separate model for the qualifying and matching host response so as not to potentially break anything?

Yes, good idea, and we will create a new issue to deprecate all obsolete models from the proof-of-concept implementation.

> Also where will be the user email be accessible from?

Since this context provider is a child of the `Auth0Context.Provider>` the `useAuth0` hook will be available within both `<HostDashboardContext.Provider>` as well as its client components communicating via the hook.

**Pull request merged**